From a62b80e2d9fe9bed57c36b427315c61b0bacb520 Mon Sep 17 00:00:00 2001
From: neomilium <neomilium@gmail.com>
Date: Sat, 13 Mar 2010 00:32:46 +0100
Subject: [PATCH 5/8] Enhance date and time formarting
  - in makeDateISO() output will now looks like 2010-03-13 instead of 2010_03_13 (ISO 8601 defines date this way)
  - video_hour's format is now 00h42m42 (was: 00:42:42) (To prevent Windows and Samba from issue due to ':' which is not allowed in filename.)

---
 canal.py |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/canal.py b/canal.py
index b4f0c4f..4b73f99 100755
--- a/canal.py
+++ b/canal.py
@@ -117,7 +117,9 @@ def buildURLdico(currentTvShow, nb_video, quality):
                 video_url=str(debitnodelist[0].firstChild.nodeValue)
                 video_date = makeDateISO(str(datevideo.firstChild.nodeValue))
                 video_hour = str(hournodelist[0].firstChild.nodeValue)
-                video_date = video_date+"_"+video_hour
+                video_hour = video_hour.replace(":", "h", 1)
+                video_hour = video_hour.replace(":", "m", 1)
+                video_date = video_date + "_" + video_hour
                 if video_url.find("rtmp")>-1:
                     URLdico[video_url]=(video_date, video_rubrique)
                     print_verbose(" + " + video_url)
@@ -128,7 +130,7 @@ def makeDateISO(madate):
     madateISO.append(b[2])
     madateISO.append(b[1])
     madateISO.append(b[0])
-    return '_'.join(madateISO)
+    return '-'.join(madateISO)
 
 
 def downloadURL(url, verbose, overwrite, force, rename):
-- 
1.6.3.3

